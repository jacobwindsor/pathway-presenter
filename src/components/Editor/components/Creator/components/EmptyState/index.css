.empty-creator-state {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 1.6rem;
    width: 60%;
    height: 60%;
}

.empty-creator-state .clickable {
    position: absolute;
    /* 100% - padding - padding - borderWidth - borderWidth - parentPadding - parentPadding */
    height: calc(100% - 1.6rem - 1.6rem - 4px - 4px - 1.6rem - 1.6rem);
    width: calc(100% - 1.6rem - 1.6rem - 4px - 4px - 1.6rem - 1.6rem);
    cursor: pointer;
    text-align: center;
    border: 4px dashed rgb(232, 232, 232);
    border-radius: 1.2rem;
    padding: 1.6rem;
}

.empty-creator-state .clickable .content {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}