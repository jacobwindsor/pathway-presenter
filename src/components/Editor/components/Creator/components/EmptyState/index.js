import React from 'react';
import PropTypes from 'prop-types';
import Paper from 'material-ui/Paper';
import StartFlag from '../../../../../../assets/start-flag.svg';
import './index.css';

const EmptyState = props => {
    const { handleClick } = props;
    return (
        <Paper className="empty-creator-state">
            <div className="clickable" onClick={handleClick}>
                <div className="content">
                    <img src={StartFlag} alt="Start flag" />
                    <h1>Get started</h1>
                    <h3><i>Click here to add a slide</i></h3>
                </div>
            </div>
        </Paper>
    )
};

EmptyState.propTypes = {
    handleClick: PropTypes.func.isRequired,
};

export default EmptyState