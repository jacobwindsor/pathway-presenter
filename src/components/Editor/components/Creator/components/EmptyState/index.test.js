import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EmptyState from './index';

describe('EmptyState. Shows an empty state dialog when there are no slides', () => {
  it('should render correctly', () => {
    const comp = shallow(<EmptyState handleClick={() => null} />);
    expect(toJson(comp)).toMatchSnapshot();
  });

  it('should fire handleClick', () => {
    const firer = jest.fn();
    const comp = shallow(<EmptyState handleClick={firer} />);

    comp.find('.clickable').first().simulate('touchTap');
    expect(firer).toHaveBeenCalledTimes(1);
  });
});
