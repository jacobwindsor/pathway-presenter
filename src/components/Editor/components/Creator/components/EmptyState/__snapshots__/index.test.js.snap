// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState. Shows an empty state dialog when there are no slides should render correctly 1`] = `
<Paper
  circle={false}
  className="empty-creator-state"
  rounded={true}
  transitionEnabled={true}
  zDepth={1}
>
  <div
    className="clickable"
    onTouchTap={[Function]}
  >
    <div
      className="content"
    >
      <img
        alt="Start flag"
        src="start-flag.svg"
      />
      <h1>
        Get started
      </h1>
      <h3>
        <i>
          Click here to add a slide
        </i>
      </h3>
    </div>
  </div>
</Paper>
`;
